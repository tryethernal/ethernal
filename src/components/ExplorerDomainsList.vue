<template>
    <v-card outlined :disabled="disabled">
        <New-Explorer-Domain-Modal ref="newExplorerDomainModal" />
        <v-card-text>
            <v-alert v-if="disabled" text type="warning">Upgrade your plan to activate domain aliases.</v-alert>
            <v-list v-if="domains.length" class="py-0 mb-4">
                <Explorer-Domain @deleted="reloadDomains()" v-for="(domain, idx) in domains" :domain="domain" :key="idx" />
            </v-list>
            <div v-else class="mb-2">
                No aliases yet.
            </div>
            <div>
                <v-btn :disabled="loading" color="primary" @click.stop="openNewExplorerDomainModal()">Add Domain Alias</v-btn>
            </div>
        </v-card-text>
    </v-card>
</template>

<script>
import ExplorerDomain from './ExplorerDomain';
import NewExplorerDomainModal from './NewExplorerDomainModal.vue';

export default {
    name: 'ExplorerDomainsList',
    props: ['explorer', 'disabled'],
    components: {
        ExplorerDomain,
        NewExplorerDomainModal
    },
    data: () => ({
        loading: false,
        domains: []
    }),
    mounted() {
        this.domains = this.explorer.domains;
    },
    methods: {
        reloadDomains() {
            this.server.getExplorer(this.explorer.id)
                .then(({ data }) => this.domains = data.domains)
                .catch(console.log);
        },
        openNewExplorerDomainModal() {
            this.$refs.newExplorerDomainModal
                .open({ explorer: this.explorer })
                .then(refresh => {
                    if (refresh) this.reloadDomains();
                });
        },
        showDnsInfo(domain) {
            this.$refs.newExplorerDomainModal
                .open({ onlyDnsInfo: true, domain });
        }
    },
    computed: {
    }
}
</script>
