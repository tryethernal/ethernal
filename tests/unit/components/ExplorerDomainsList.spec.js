import flushPromises from 'flush-promises';
import MockHelper from '../MockHelper';

import ExplorerDomainsList from '@/components/ExplorerDomainsList.vue';

let helper;
beforeEach(() => {
    jest.clearAllMocks()
    helper = new MockHelper();
});

describe('ExplorerDomainsList.vue', () => {
    it('Should display domains list', async () => {
        const wrapper = helper.mountFn(ExplorerDomainsList, {
            stubs: ['Explorer-Domain', 'New-Explorer-Domain-Modal'],
            propsData: {
                disabled: false,
                explorer: {
                    domains: [{ id: 1 }, { id: 2 }]
                }
            }
        });
        await flushPromises();

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Should be disabled', async () => {
        const wrapper = helper.mountFn(ExplorerDomainsList, {
            stubs: ['Explorer-Domain', 'New-Explorer-Domain-Modal'],
            propsData: {
                disabled: true,
                explorer: {
                    domains: [{ id: 1 }, { id: 2 }]
                }
            }
        });
        await flushPromises();

        expect(wrapper.html()).toMatchSnapshot();
    });
});
