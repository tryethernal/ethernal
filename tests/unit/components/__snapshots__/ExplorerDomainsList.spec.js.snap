// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExplorerDomainsList.vue Should be disabled 1`] = `
<div class="v-card v-card--disabled v-sheet v-sheet--outlined theme--light">
  <new-explorer-domain-modal-stub></new-explorer-domain-modal-stub>
  <div class="v-card__text">
    <div role="alert" class="v-alert v-sheet theme--dark v-alert--text warning--text">
      <div class="v-alert__wrapper"><i aria-hidden="true" class="v-icon notranslate v-alert__icon mdi mdi-exclamation theme--dark warning--text"></i>
        <div class="v-alert__content">Upgrade your plan to activate domain aliases.</div>
      </div>
    </div>
    <div role="list" class="v-list py-0 mb-4 v-sheet theme--light">
      <explorer-domain-stub domain="[object Object]"></explorer-domain-stub>
      <explorer-domain-stub domain="[object Object]"></explorer-domain-stub>
    </div>
    <div><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default primary"><span class="v-btn__content">Add Domain Alias</span></button></div>
  </div>
</div>
`;

exports[`ExplorerDomainsList.vue Should display domains list 1`] = `
<div class="v-card v-sheet v-sheet--outlined theme--light">
  <new-explorer-domain-modal-stub></new-explorer-domain-modal-stub>
  <div class="v-card__text">
    <!---->
    <div role="list" class="v-list py-0 mb-4 v-sheet theme--light">
      <explorer-domain-stub domain="[object Object]"></explorer-domain-stub>
      <explorer-domain-stub domain="[object Object]"></explorer-domain-stub>
    </div>
    <div><button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default primary"><span class="v-btn__content">Add Domain Alias</span></button></div>
  </div>
</div>
`;
